test/integration/relations.test.js

```diff
@@ -119,101 +136,78 @@ describe('Relations', function () {
                     });
             });
+        });
 
-            it('tags.slug equals "animal" and posts_tags.sort_order is 0 and featured is false', function () {
-                // where primary tag is "animal"
-                const mongoJSON = {$and: [
-                    {
-                        $and: [
-                            {
-                                'tags.slug': 'animal'
-                            },
-                            {
-                                'posts_tags.sort_order': 0
-                            }
-                        ]
-                    },
-                    {
-                        featured: false
-                    }
-                ]};
+        describe('AND $and', function () {
+            it('tags.slug is animal and classic', function () {
+                const mongoJSON = {
+                    $and: [
+                        {
+                            'tags.slug': 'animal'
+                        },
+                        {
+                            'tags.slug': 'classic'
+                        }
+                    ]
+                };
 
                 const query = makeQuery(mongoJSON);
 
                 return query
                     .select()
                     .then((result) => {
-                        result.should.be.an.Array().with.lengthOf(1);
+                        result.should.be.an.Array().with.lengthOf(2);
                     });
             });
 
-            it('tags.slug equals "animal" and posts_tags.sort_order is 0 OR author_id is 1', function () {
-                const mongoJSON = {$or: [
-                    {
-                        $and: [
-                            {
-                                'tags.slug': 'animal'
-                            },
-                            {
-                                'posts_tags.sort_order': 0
-                            },
-                            {
-                                featured: false
-                            }
-                        ]
-                    },
-                    {author_id: 1}
-                ]};
+            it('tags.slug is hash-internal and tags.visibility is private', function () {
+                const mongoJSON = {
+                    $and: [
+                        {
+                            'tags.slug': 'hash-internal'
+                        },
+                        {
+                            'tags.visibility': 'internal'
+                        }
+                    ]
+                };
 
                 const query = makeQuery(mongoJSON);
 
                 return query
                     .select()
                     .then((result) => {
-                        result.should.be.an.Array().with.lengthOf(6);
+                        result.should.be.an.Array().with.lengthOf(1);
+                        result[0].title.should.equal('has internal tag');
```

Comment:
Started using `equal` comparison in tests that I've touched to make sure the results are actually correct (experienced some false positives when just checking a count)